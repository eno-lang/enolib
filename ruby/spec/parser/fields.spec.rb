# frozen_string_literal: true

input = <<~DOC.strip
key:value
key: value
key: more value
    key    :    value
DOC

describe Enolib::Parser do
  describe 'Fields' do
    context = Enolib::Context.new(input)

    it 'parses as expected' do
      expect(context.document).to match_snapshot
    end
  end
end
