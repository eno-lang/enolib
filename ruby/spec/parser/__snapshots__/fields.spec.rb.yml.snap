---
"./spec/tokenizer/fields.spec.rb[1:1:1]":
- :index: 0
  :line: 0
  :key: name
  :ranges:
    :key:
    - 0
    - 4
    :field_operator:
    - 4
    - 5
    :value:
    - 5
    - 10
  :type: :field
  :value: value
  :length: 10
- :index: 11
  :line: 1
  :key: name
  :ranges:
    :key:
    - 0
    - 4
    :field_operator:
    - 4
    - 5
    :value:
    - 6
    - 11
  :type: :field
  :value: value
  :length: 11
- :index: 23
  :line: 2
  :key: name
  :ranges:
    :key:
    - 0
    - 4
    :field_operator:
    - 4
    - 5
    :value:
    - 6
    - 16
  :type: :field
  :value: more value
  :length: 16
- :index: 40
  :line: 3
  :key: name
  :ranges:
    :key:
    - 4
    - 8
    :field_operator:
    - 12
    - 13
    :value:
    - 17
    - 22
  :type: :field
  :value: value
  :length: 22
"./spec/parser/fields.spec.rb[1:1:1]": &1
  :elements:
  - :index: 0
    :line: 0
    :ranges:
      :line:
      - 0
      - 9
      :field_operator:
      - 3
      - 4
      :key:
      - 0
      - 3
      :value:
      - 4
      - 9
    :key: key
    :type: :field
    :value: value
    :parent: *1
  - :index: 10
    :line: 1
    :ranges:
      :line:
      - 10
      - 20
      :field_operator:
      - 13
      - 14
      :key:
      - 10
      - 13
      :value:
      - 15
      - 20
    :key: key
    :type: :field
    :value: value
    :parent: *1
  - :index: 21
    :line: 2
    :ranges:
      :line:
      - 21
      - 36
      :field_operator:
      - 24
      - 25
      :key:
      - 21
      - 24
      :value:
      - 26
      - 36
    :key: key
    :type: :field
    :value: more value
    :parent: *1
  - :index: 37
    :line: 3
    :ranges:
      :line:
      - 37
      - 58
      :field_operator:
      - 48
      - 49
      :key:
      - 41
      - 44
      :value:
      - 53
      - 58
    :key: key
    :type: :field
    :value: value
    :parent: *1
  :type: :document
